app/assets/javascripts/utils/media_wiki_classes.js

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable no-console */
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable no-await-in-loop */
+import request from './request';
+import { PageAssessmentGrades } from './article_finder_language_mappings';
+import { stringify } from 'query-string';
+import { chunk } from 'lodash-es';
+
+// this function takes in a wiki url and the revisions of that particular wiki
+// it then finds the page assessments of all of the articles, merges them together
+// and returns an object in the form of {revision_id: {rating, pretty_rating, rating_num}, ...}
+const fetchClassFromRevisionsOfWiki = async (wiki_url, revisionsOfWiki) => {
+  // remove duplicates -> each article occurs only once after this
+  const prefix = `https://${wiki_url}`;
+  const API_URL = `${prefix}/w/api.php`;
+  // eslint-disable-next-line no-console
+  console.log(`Fetching page assessments from ${wiki_url}`);
+
+  const uniqueArticles = [...new Set(revisionsOfWiki.filter(revision => revision.ns === 0).map(revision => revision.title))];
+  const allRatings = [];
+  const chunks = chunk(uniqueArticles, 50);
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const uniqueArticlesChunk of chunks) {
+    const params = {
```

Comment:
The code for making an individual request can probably be extracted to a separate function here.

Reply:
Well, its not related to this PR, but it does fixes a problem - the wikidata base url is https://www.wikidata.org The function earlier evaluation it to https://wikidata.org Clicking on the diff viewer for a wikidata item earlier lead to a 301 (Moved Permanently) error.

If you think this change should be made in another PR, I'm fine with that as well.